dockstore-webservice/src/main/java/io/dockstore/webservice/core/AppTool.java

```diff
@@ -54,12 +69,14 @@ public EntryTypeMetadata getEntryTypeMetadata() {
     }
 
     @Override
-    public Entry getParentEntry() {
+    @OneToOne
+    @Schema(hidden = true)
```

Comment:
I think hiding these parent relationships was necessary for getting rid of the odd `EntryObjectObject` schemas in `openapi.yaml`

Reply:
That's a much simpler and obvious-solution-in-hindsight than my refactoring the code to get rid of self-referencing generics. Even if we don't take this whole PR, can we just take this?